src/CodeAnalysis/CSharp/CSharpCodeAnalysisDiagnosticAnalyzers.csproj

```diff
@@ -42,6 +42,7 @@
   <ItemGroup>
     <Compile Include="FixAnalyzers\CSharpFixerWithFixAllAnalyzer.cs" />
     <Compile Include="MetaAnalyzers\CSharpDiagnosticDescriptorCreationAnalyzer.cs" />
+    <Compile Include="MetaAnalyzers\CSharpImmutableObjectMethodAnalyzer.cs" />
```

Comment:
This analyzer would probably be better outside the MetaAnalyzers folder - the analyzers in that folder are analyzers for the DiagnosticAnalyzer API itself (or the fix API) whereas this analyzer is about the rest of Roslyn. I'd just put it in the root folder itself.